Client/Frontend/Toolbar+URLBar/OverlayModeManager.swift

```diff
@@ -4,42 +4,70 @@
 
 import Foundation
 
-protocol OverlayModeManager {
+protocol OverlayStateProtocol {
     var inOverlayMode: Bool { get }
-    func enterOverlayMode(_ locationText: String?, pasted: Bool, search: Bool)
-    func leaveOverlayMode(didCancel cancel: Bool)
 }
 
-extension OverlayModeManager {
-    func enterOverlayMode(_ locationText: String? = nil, pasted: Bool = false, search: Bool = true) {
-        enterOverlayMode(locationText, pasted: pasted, search: search)
-    }
+protocol OverlayModeManager: OverlayStateProtocol {
+    /// Set URLBar which is not available when the manager is created
+    /// - Parameter urlBarView: URLBar that contains textfield which open and dismiss the keyboard
+    func setURLBar(urlBarView: URLBarViewProtocol)
 
-    func leaveOverlayMode(didCancel cancel: Bool = false) {
-        leaveOverlayMode(didCancel: cancel)
-    }
+    /// Enter overlay mode with paste content
+    /// - Parameter pasteContent: String with the content to paste on the search
+    func openSearch(with pasteContent: String)
+
+    /// Enter overlay mode when opening a new tab
+    /// - Parameters:
+    ///   - locationText: String with initial search text
+    ///   - url: Tab url to determine if is the url is homepage or nil
+    func openNewTab(_ locationText: String?, url: URL?)
+
+    /// Leave overlay mode when user finish edition, either pressing the go button, enter etc
+    /// - Parameter shouldCancelLoading: Bool value determine if the loading animation of the current search should be canceled
+    func finishEdition(shouldCancelLoading: Bool)
```

Comment:
I think 'editing' would be better here than 'edition'. What do you think @adudenamedruby ?
```suggestion
    /// Leave overlay mode when user finish editing, either pressing the go button, enter etc
    /// - Parameter shouldCancelLoading: Bool value determine if the loading animation of the current search should be canceled
    func finishEditing(shouldCancelLoading: Bool)
```

Reply:
I like it, I don't get any other feedback I will like to merge as it is and leave this change for the next ticket

Reply:
Yeah, "editing" works better. But I'm good to leave it and fix next pass